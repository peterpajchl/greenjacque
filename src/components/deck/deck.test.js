import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import Deck from './deck';
import {
  assembleDeck
} from '../../common/common';
import Card from '../../common/card';

Enzyme.configure({ adapter: new Adapter() });

describe('<Deck /> component', () => {
  const deck = assembleDeck();

  it('should match the snapshot', () => {
    const handleDraw = sinon.spy();
    const handleShuffle = sinon.spy();
    const wrapper = shallow(<Deck cards={deck} handleDraw={handleDraw} handleShuffle={handleShuffle} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render 52 cards when given full deck', () => {
    const handleDraw = sinon.spy();
    const handleShuffle = sinon.spy();
    const wrapper = shallow(<Deck cards={deck} handleDraw={handleDraw} handleShuffle={handleShuffle} />);
    expect(wrapper.find('.deck-card').length).toEqual(52);
  });

  it('should render shuffle button', () => {
    const handleDraw = sinon.spy();
    const handleShuffle = sinon.spy();
    const wrapper = shallow(<Deck cards={deck} handleDraw={handleDraw} handleShuffle={handleShuffle} />);
    const button = wrapper.find('button[type="button"].control');
    expect(button.length).toEqual(1);
    expect(button.text()).toEqual('Shuffle');
  });

  it('should render draw button', () => {
    const handleDraw = sinon.spy();
    const handleShuffle = sinon.spy();
    const wrapper = shallow(<Deck cards={deck} handleDraw={handleDraw} handleShuffle={handleShuffle} />);
    const button = wrapper.find('button[type="button"].draw-card__submit');
    expect(button.length).toEqual(1);
    expect(button.text()).toEqual('Draw');
  });

  it('should allow to call shuffle', () => {
    const handleDraw = sinon.spy();
    const handleShuffle = sinon.spy();
    const wrapper = shallow(<Deck cards={deck} handleDraw={handleDraw} handleShuffle={handleShuffle} />);
    const button = wrapper.find('button[type="button"].control');
    button.simulate('click');
    expect(handleShuffle).toHaveProperty('callCount', 1);
  });

  it('should allow to call draw', () => {
    const handleDraw = sinon.spy();
    const handleShuffle = sinon.spy();
    const wrapper = shallow(<Deck cards={deck} handleDraw={handleDraw} handleShuffle={handleShuffle} />);
    const button = wrapper.find('button[type="button"].draw-card__submit');
    button.simulate('click');
    expect(handleDraw).toHaveProperty('callCount', 1);
  });

});