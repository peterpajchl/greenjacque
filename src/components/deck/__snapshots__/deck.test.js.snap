// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Deck /> component should match the snapshot 1`] = `"<div class=\\"main-body__block main-body__block--deck\\"><div class=\\"body-header\\"><h3 class=\\"body-header__title\\">Deck</h3><div class=\\"body-header__counter\\">52/52</div></div><ul class=\\"cards-in-deck__list\\"><li class=\\"deck-card\\" style=\\"left:12.75px;top:0.5px;z-index:51\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"2C\\"/></li><li class=\\"deck-card\\" style=\\"left:12.5px;top:1px;z-index:50\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"3C\\"/></li><li class=\\"deck-card\\" style=\\"left:12.25px;top:1.5px;z-index:49\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"4C\\"/></li><li class=\\"deck-card\\" style=\\"left:12px;top:2px;z-index:48\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"5C\\"/></li><li class=\\"deck-card\\" style=\\"left:11.75px;top:2.5px;z-index:47\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"6C\\"/></li><li class=\\"deck-card\\" style=\\"left:11.5px;top:3px;z-index:46\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"7C\\"/></li><li class=\\"deck-card\\" style=\\"left:11.25px;top:3.5px;z-index:45\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"8C\\"/></li><li class=\\"deck-card\\" style=\\"left:11px;top:4px;z-index:44\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"9C\\"/></li><li class=\\"deck-card\\" style=\\"left:10.75px;top:4.5px;z-index:43\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"10C\\"/></li><li class=\\"deck-card\\" style=\\"left:10.5px;top:5px;z-index:42\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"JC\\"/></li><li class=\\"deck-card\\" style=\\"left:10.25px;top:5.5px;z-index:41\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"QC\\"/></li><li class=\\"deck-card\\" style=\\"left:10px;top:6px;z-index:40\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"KC\\"/></li><li class=\\"deck-card\\" style=\\"left:9.75px;top:6.5px;z-index:39\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"AC\\"/></li><li class=\\"deck-card\\" style=\\"left:9.5px;top:7px;z-index:38\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"2S\\"/></li><li class=\\"deck-card\\" style=\\"left:9.25px;top:7.5px;z-index:37\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"3S\\"/></li><li class=\\"deck-card\\" style=\\"left:9px;top:8px;z-index:36\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"4S\\"/></li><li class=\\"deck-card\\" style=\\"left:8.75px;top:8.5px;z-index:35\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"5S\\"/></li><li class=\\"deck-card\\" style=\\"left:8.5px;top:9px;z-index:34\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"6S\\"/></li><li class=\\"deck-card\\" style=\\"left:8.25px;top:9.5px;z-index:33\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"7S\\"/></li><li class=\\"deck-card\\" style=\\"left:8px;top:10px;z-index:32\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"8S\\"/></li><li class=\\"deck-card\\" style=\\"left:7.75px;top:10.5px;z-index:31\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"9S\\"/></li><li class=\\"deck-card\\" style=\\"left:7.5px;top:11px;z-index:30\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"10S\\"/></li><li class=\\"deck-card\\" style=\\"left:7.25px;top:11.5px;z-index:29\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"JS\\"/></li><li class=\\"deck-card\\" style=\\"left:7px;top:12px;z-index:28\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"QS\\"/></li><li class=\\"deck-card\\" style=\\"left:6.75px;top:12.5px;z-index:27\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"KS\\"/></li><li class=\\"deck-card\\" style=\\"left:6.5px;top:13px;z-index:26\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"AS\\"/></li><li class=\\"deck-card\\" style=\\"left:6.25px;top:13.5px;z-index:25\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"2H\\"/></li><li class=\\"deck-card\\" style=\\"left:6px;top:14px;z-index:24\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"3H\\"/></li><li class=\\"deck-card\\" style=\\"left:5.75px;top:14.5px;z-index:23\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"4H\\"/></li><li class=\\"deck-card\\" style=\\"left:5.5px;top:15px;z-index:22\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"5H\\"/></li><li class=\\"deck-card\\" style=\\"left:5.25px;top:15.5px;z-index:21\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"6H\\"/></li><li class=\\"deck-card\\" style=\\"left:5px;top:16px;z-index:20\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"7H\\"/></li><li class=\\"deck-card\\" style=\\"left:4.75px;top:16.5px;z-index:19\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"8H\\"/></li><li class=\\"deck-card\\" style=\\"left:4.5px;top:17px;z-index:18\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"9H\\"/></li><li class=\\"deck-card\\" style=\\"left:4.25px;top:17.5px;z-index:17\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"10H\\"/></li><li class=\\"deck-card\\" style=\\"left:4px;top:18px;z-index:16\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"JH\\"/></li><li class=\\"deck-card\\" style=\\"left:3.75px;top:18.5px;z-index:15\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"QH\\"/></li><li class=\\"deck-card\\" style=\\"left:3.5px;top:19px;z-index:14\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"KH\\"/></li><li class=\\"deck-card\\" style=\\"left:3.25px;top:19.5px;z-index:13\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"AH\\"/></li><li class=\\"deck-card\\" style=\\"left:3px;top:20px;z-index:12\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"2D\\"/></li><li class=\\"deck-card\\" style=\\"left:2.75px;top:20.5px;z-index:11\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"3D\\"/></li><li class=\\"deck-card\\" style=\\"left:2.5px;top:21px;z-index:10\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"4D\\"/></li><li class=\\"deck-card\\" style=\\"left:2.25px;top:21.5px;z-index:9\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"5D\\"/></li><li class=\\"deck-card\\" style=\\"left:2px;top:22px;z-index:8\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"6D\\"/></li><li class=\\"deck-card\\" style=\\"left:1.75px;top:22.5px;z-index:7\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"7D\\"/></li><li class=\\"deck-card\\" style=\\"left:1.5px;top:23px;z-index:6\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"8D\\"/></li><li class=\\"deck-card\\" style=\\"left:1.25px;top:23.5px;z-index:5\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"9D\\"/></li><li class=\\"deck-card\\" style=\\"left:1px;top:24px;z-index:4\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"10D\\"/></li><li class=\\"deck-card\\" style=\\"left:0.75px;top:24.5px;z-index:3\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"JD\\"/></li><li class=\\"deck-card\\" style=\\"left:0.5px;top:25px;z-index:2\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"QD\\"/></li><li class=\\"deck-card\\" style=\\"left:0.25px;top:25.5px;z-index:1\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"KD\\"/></li><li class=\\"deck-card\\" style=\\"left:0px;top:26px;z-index:0\\"><img class=\\"card-deck__image\\" width=\\"100%\\" src=\\"./assets/cards/1B.svg\\" alt=\\"AD\\"/></li></ul><div class=\\"card-controls\\"><button class=\\"btn control\\" type=\\"button\\">Shuffle</button><form method=\\"GET\\" action=\\"/\\" class=\\"draw-card control\\"><label for=\\"draw-card-input\\" class=\\"sr-only\\">Enter number of cards to draw</label><input type=\\"number\\" id=\\"draw-card-input\\" class=\\"draw-card__input\\" min=\\"1\\" max=\\"52\\" value=\\"1\\"/><button class=\\"draw-card__submit btn\\" type=\\"button\\">Draw</button></form></div></div>"`;
